### ISSUE TITLE
question: living Cursive seems to locks std out  

### ISSUE THREAD

**Issue by douglasduteil** [gyscos/Cursive#180]
Hi @gyscos 

I'm new to Rust and I'm inspired by your work on Cursive :+1: 
Can you help me understand why any new Cursice instance is blocking stdout ?

Here is two cases I compared :

```rust
extern crate cursive;

use cursive::Cursive;
use cursive::views::{Dialog, TextView};

fn main() {
  println!("A");

  let mut siv = Cursive::new();

  println!("B");

  siv.add_layer(
    Dialog::around(TextView::new("Hello Dialog!"))
      .title("Cursive")
      .button("Quit", |s| s.quit()),
  );

  println!("C");
}

# $ cargo run
# A
```

Compared to 

```rust
extern crate cursive;

use cursive::Cursive;
use cursive::views::{Dialog, TextView};

fn main() {
  println!("A");

  {
    let mut siv = Cursive::new();

    println!("B");

    siv.add_layer(
      Dialog::around(TextView::new("Hello Dialog!"))
        .title("Cursive")
        .button("Quit", |s| s.quit()),
    );
  }

  println!("C");
}

# $ cargo run
# A
# C
```

Apparently as long as a Cursive instance is alive, we can't print to stdout.
Do I get it right ?

Thanks

**Comment by hellow554** [comment]
See the [documentation](https://docs.rs/cursive/0.7.5/cursive/#debugging)

How else is the gui supposed to display itself? (hint: it doesn't use stdin) ;)

**Comment by gyscos** [comment]
Correct: currently, `Cursive` initialize the terminal on creation, and resets it to the previous value on drop.

It might be possible to only take control of the terminal during the `run` method; I'll need to check if it doesn't break anything (it breaks `Cursive::screen_size` for instance).

**Comment by douglasduteil** [comment]
Alright.

As you say @gyscos , I expected that Cursive takes full control only when running the `run` function.

Thanks for your responses @hellow554 @gyscos .
I'm guessing that it's the `backend` from [`backend::Concrete::init();`](https://github.com/gyscos/Cursive/blob/master/src/backend/termion.rs#L145-L148) doing right ?

I would like this change to be made but... 
Do you think it's worth it ?
I will try to make a PR on Friday if it is 😄 

Regards

**Comment by gyscos** [comment]
Some methods like `Cursive::screen_size` expects the backend to be initialized even before `run`: indeed, some users use it when creating the view. Moving the initialization in the `run` would break their applications.
It may be fixeable - I don't think the terminal *needs* to be initialized to retrieve its size, so it may be a ncurses limitation.

In addition, some users do not use `Cursive::run`, but run the event loop manually with `Cursive::step`.

I have a design in mind for that, but that's only if we find a way to minimize breakage.
